.github/instructions/generic/ember_hbs.instructions.md

```diff
@@ -0,0 +1,110 @@
+---
+applyTo: "vault/ui/**/*.hbs"
+description: "HashiCorp Ember.js UI Handlebars template coding standards"
+---
+
+# HashiCorp Ember.js Handlebars Template Guidelines
+
+This document provides Handlebars template coding standards for HashiCorp Ember.js UI applications.
+
+> **Note**: For general project context, framework information, and repository structure, see `ember_general.instructions.md`.
+
+## Template Best Practices
+- Check truthiness of arrays directly instead of using `.length` property
+- Use string interpolation `"prefix/{{value}}"` instead of `{{concat}}` helper  
+- Remove unnecessary quotes around dynamic component arguments
+- Use `Hds::Link::Inline` for external documentation links instead of `<button>` elements
+- Make `selected` attributes dynamic rather than static values - warn if static values are used
+- Refactor conditionals to wrap content rather than entire elements when possible
+- Avoid inline `style` attributes - define CSS classes in `.scss` files instead
+- Place `data-test-*` selectors as the last attribute on elements
+- Remove quotes around dynamic data attributes: `data-test-id={{value}}` not `data-test-id="{{value}}"`
+
+Examples:
+```handlebars
+{{!-- Good: direct array check --}}
+{{#if this.model.allowed_roles}}
+
+{{!-- Bad: unnecessary .length check --}}
+{{#if (gt this.model.allowed_roles.length 0)}}
+
+{{!-- Good: string interpolation --}}
+@secret="role/{{@model.id}}"
+
+{{!-- Bad: concat helper --}}
+@secret={{concat "role/" @model.id}}
+
+{{!-- Good: conditional content, not element --}}
+<PH.Title>{{if this.version.isEnterprise "Enterprise" "Community"}} features</PH.Title>
+
+{{!-- Bad: conditional wrapping entire element --}}
+{{#if this.version.isEnterprise}}
+  <PH.Title>Enterprise features</PH.Title>
+{{else}}
+  <PH.Title>Community features</PH.Title>
+{{/if}}
+
+{{!-- Good: CSS classes instead of inline styles --}}
+<Hds::Button @text="Save" class="custom-button" data-test-save />
+
+{{!-- Bad: inline style attribute --}}
+<Hds::Button @text="Save" style="margin-top: 10px;" data-test-save />
```

Comment:
The syntax I've [seen](https://github.com/hashicorp/vault/blob/main/ui/app/components/clients/charts/vertical-bar-basic.hbs#L78) for this is also `{{style ...}}`